CodeCoverageQualityGate/CodeCoverageQualityService.ts

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2023 Parasoft Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as tl from 'azure-pipelines-task-lib/task';
+
+export const enum TypeEnum {
+    OVERALL = "Overall",
+    MODIFIED = "Modified",
+}
+
+export const enum BuildStatusEnum {
+    FAILED = "Failed",
+    UNSTABLE = "Unstable"
+}
+
+export class CodeCoverageQualityService {
+    readonly typeString: string;
+    readonly thresholdString: string;
+    readonly buildStatusString: string;
+
+    type: TypeEnum;
+    threshold: number;
+    buildStatus: BuildStatusEnum;
+
+    constructor() {
+        this.typeString = tl.getInput('type') || '';
+        this.thresholdString = tl.getInput('threshold') || '';
+        this.buildStatusString = tl.getInput('buildStatus') || '';
+
+        this.threshold = parseFloat(this.thresholdString || '0.0');
+
+        if (isNaN(this.threshold)) {
+            tl.warning('Illegal threshold value \'' + this.thresholdString + '\', using default value 0.0');
```

Comment:
Change Illegal to Invalid.

Reply:
Use the format string like: `Invalid threshold value '${this.thresholdString}', using default value 0.0`. The same change need to be done for other places.